gh-138122: Add some screenshots to the profiling.sampling docs (#142676)
python/cpython: Doc/library/profiling.sampling.rst

Opcode information appears in several output formats:

- **Flame graphs**: Hovering over a frame displays a tooltip with bytecode

>   instruction breakdown, showing which opcodes consumed time in that function

Reviewer: ```suggestion
- **Flame graphs**: Hovering over a frame displays a tooltip with a bytecode
  instruction breakdown, showing which opcodes consumed time in that function
```

   python -m profiling.sampling run --flamegraph script.py
   python -m profiling.sampling run --flamegraph -o profile.html script.py

.. figure:: tachyon-flamegraph.png
   :alt: Tachyon interactive flame graph
   :align: center
   :width: 100%

   The flame graph visualization shows call stacks as nested rectangles, with
   width proportional to time spent. The sidebar displays runtime statistics,
   GIL metrics, and hotspot functions.

.. only:: html

>    `Try the interactive example <../_static/tachyon-example-flamegraph.html>`__.

Reviewer: ```suggestion
   `Try the interactive example <../_static/tachyon-example-flamegraph.html>`__!
```
Little suggestion.

For this reason, the :option:`--mode` option is not available with Gecko format;
all relevant data is captured automatically.

.. figure:: tachyon-gecko-calltree.png
   :alt: Firefox Profiler Call Tree view
   :align: center
   :width: 100%

   The Call Tree view shows the complete call hierarchy with sample counts
   and percentages. The sidebar displays detailed statistics for the
   selected function including running time and sample distribution.

.. figure:: tachyon-gecko-flamegraph.png
   :alt: Firefox Profiler Flame Graph view
   :align: center
   :width: 100%

   The Flame Graph visualization shows call stacks as nested rectangles.
   Functions like ``worker_a``, ``level_1_analyze``, ``level_2_process``,

>    and ``level_4_compute`` are visible in the call hierarchy.

Reviewer: ```suggestion
   Functions names are visible in the call hierarchy.
```
I suggest simplifying this, it would make sense IMO if we had the script we were profiling above.

- **Dark/light theme**: toggle with preference saved across sessions
- **Line linking**: click line numbers to create shareable URLs

When opcode-level profiling is enabled with :option:`--opcodes`, each hot line
can be expanded to show which bytecode instructions consumed time:

.. figure:: tachyon-heatmap-with-opcodes.png
   :alt: Heatmap with expanded bytecode panel
   :align: center
   :width: 100%

   Expanding a hot line reveals the bytecode instructions executed, including
   specialized variants. The panel shows sample counts per instruction and the
   overall specialization percentage for the line.

.. only:: html

>    `Try the interactive example <../_static/tachyon-example-heatmap.html>`__.

Reviewer: ```suggestion
   `Try the interactive example <../_static/tachyon-example-heatmap.html>`__!
```

   python -m profiling.sampling run --live script.py
   python -m profiling.sampling attach --live 12345

.. figure:: tachyon-live-mode-2.gif
   :alt: Tachyon live mode showing all threads
   :align: center
   :width: 100%

   Live mode displays real-time profiling statistics showing combined

>    statistics from multiple threads in a multi-threaded application.

Reviewer: ```suggestion
   Live mode displays real-time profiling statistics, showing combined
   data from multiple threads in a multi-threaded application.
```
To avoid the repeated "statistics".

function. This panel displays which bytecode instructions are executing most
frequently, including specialized variants and their base opcodes.

.. figure:: tachyon-live-mode-1.gif
   :alt: Tachyon live mode with opcode panel
   :align: center
   :width: 100%

   Live mode with ``--opcodes`` enabled shows an opcode panel with bytecode

>    instruction breakdown for the selected function.

Reviewer: ```suggestion
   Live mode with ``--opcodes`` enabled shows an opcode panel with a bytecode
   instruction breakdown for the selected function.
```
Grammar: missing an article.
